snake_state.py

```diff
@@ -234,6 +241,9 @@ def _grow_step(self, old_tail_pos):
 
     def _respawn_apple(self):
         self.__apple.move(self.__grid.find_free_cell())
+        
+    def _get_dist_to_apple(self, head_pos):
```

Comment:
Make this the property, instead of having the property call this method. You have full access of `self` from here, so there's no point in having arguments etc. like you have now.